src/ja/kadocomi/src/eu/kanade/tachiyomi/extension/ja/kadocomi/KadoComi.kt

```diff
@@ -0,0 +1,319 @@
+package eu.kanade.tachiyomi.extension.ja.kadocomi
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Interceptor
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.ResponseBody.Companion.toResponseBody
+import uy.kohesive.injekt.injectLazy
+import java.text.ParseException
+import java.text.SimpleDateFormat
+import java.util.Locale
+import kotlin.experimental.xor
+
+class KadoComi : HttpSource() {
+
+    override val name = "カドコミ" // KadoComi, formerly Comic Walker
+
+    override val baseUrl = "https://comic-walker.com"
+
+    private val apiUrl = "https://comic-walker.com/api"
+
+    private val cdnUrl = "https://cdn.comic-walker.com"
+
+    override val lang = "ja"
+
+    override val supportsLatest = true
+
+    private val imageDescrambler: Interceptor = Interceptor { chain ->
+        val request: Request = chain.request()
+        val urlString = request.url.toString()
+        val drmHash = request.url.fragment ?: ""
+
+        val response: Response = chain.proceed(request)
+
+        if (urlString.contains("$cdnUrl/images/") && urlString.contains("&Key-Pair-Id=")) {
+            val oldBody = response.body.bytes()
+            val descrambled = descrambleImage(oldBody, drmHash.decodeHex())
+            val newBody = descrambled.toResponseBody("image/jpeg".toMediaTypeOrNull())
+            response.newBuilder()
+                .body(newBody)
+                .build()
+        } else {
+            response
+        }
+    }
+
+    override val client = network.client.newBuilder()
+        .addNetworkInterceptor(imageDescrambler)
+        .build()
+
+    private val json: Json by injectLazy()
+
+    // ============================== Manga Details ===============================
+
+    override fun getMangaUrl(manga: SManga): String {
+        return "$baseUrl/detail/${getWorkCode(manga)}"
+    }
+
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        val url = apiUrl.toHttpUrl().newBuilder().apply {
+            addPathSegment("contents")
+            addPathSegment("details")
+            addPathSegment("work")
+            addQueryParameter("workCode", getWorkCode(manga))
+        }
+
+        return GET(url.build(), headers)
+    }
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val details = KadoComiWorkDto.let {
+            json.decodeFromString<KadoComiWorkDto>(response.body.string())
+        }
```

Comment:
```suggestion
        val details = json.decodeFromString<KadoComiWorkDto>(response.body.string())
```

Same for the others too